hbase-server/src/main/java/org/apache/hadoop/hbase/master/ServerManager.java

```diff
@@ -620,8 +620,7 @@ public synchronized boolean expireServer(final ServerName serverName) {
     }
   }
 
-  @VisibleForTesting
-  public void moveFromOnlineToDeadServers(final ServerName sn) {
+  public synchronized void moveFromOnlineToDeadServers(final ServerName sn) {
```

Comment:
Checked the code, it is used for HBCK2 so I do not think synchronized is necessary there as it is a human operation. Anyway it is fine to synchronized for now, can review it later. But please add the VisibleForTesting back, that's why we make it public, otherwise we can make it package private. This can also be polished later.